bytes_ostream.hh

```diff
@@ -268,6 +268,35 @@ public:
         write(bytes_view(reinterpret_cast<const signed char*>(ptr), size));
     }
 
+    // Writes the fragmented view
+    template<FragmentedView View>
+    void write(View& v) {
+        // copy data into currently available space
+        while (current_space_left() > 0 && !v.empty()) {
+            auto this_size = std::min(v.current_fragment().size(), size_t(current_space_left()));
+            memcpy(_current->data + _current->frag_size, v.current_fragment().begin(), this_size);
+            _current->frag_size += this_size;
+            _size += this_size;
+            v.remove_prefix(this_size);
+        }
+
+        // copy rest of the fragments one by one into newly alloc'ed chunks
+        while (!v.empty()) {
+            auto this_size = std::min(v.current_fragment().size(), size_t(max_chunk_size()));
+            std::copy_n(v.current_fragment().begin(), this_size, alloc_new(this_size));
+            v.remove_prefix(this_size);
+        }
```

Comment:
FragmentedView doesn't guarantee that `current_fragment()` is nonempty.
If `current_fragment()` is empty, this will become an infinite loop.

But why did you even add a new write primitive for this, instead of using the existing ones? I suggest
```c++
    template<FragmentedView View>
    void write(const View& v) {
        for (bytes_view f : fragment_range(v)) {
            write(f);
        }
    }
```

Adding new "low-level" methods has to be justified by some performance reasons, otherwise it adds space for bugs, like this one.